contracts/cyberpunk/src/contract.rs

```diff
@@ -297,4 +298,161 @@ mod tests {
 
         assert_eq!(denom.symbol, "FOO0");
     }
+
+    #[test]
+    fn query_denom_metadata_with_null_denom_units_works() {
```

Comment:
Nice tests, but could you put them into the same file as the `DenomMetadata` type? It doesn't really make sense to have this in the cyberpunk contract.

Reply:
Thank you for pointing that out! I rolled back the codes in the cyberpunk contract and added an integration test mod to the metadata file.